src/server/scripts/EasternKingdoms/Karazhan/instance_karazhan.cpp

```diff
@@ -118,7 +150,28 @@ class instance_karazhan : public InstanceMapScript
                     m_auiEncounter[1] = uiData;
                     break;
                 case TYPE_MAIDEN:               m_auiEncounter[2] = uiData; break;
-                case TYPE_OPTIONAL_BOSS:        m_auiEncounter[3] = uiData; break;
+                case TYPE_OPTIONAL_BOSS:
+                    m_auiEncounter[3] = uiData;
+                    if (uiData == NOT_STARTED)
+                    {
+                        ++OptionalBossCount;
+                        if (OptionalBossCount == 50)
```

Comment:
@DDuarte there are 50 trashmobs, if all of them are dows one randomboss must spawn